src/game/common/gameengine.cpp

```diff
@@ -96,20 +101,88 @@ GameEngine::GameEngine() : m_maxFPS(0), m_isQuitting(false), m_isActive(false)
 #ifdef PLATFORM_WINDOWS
     timeBeginPeriod(1);
 #endif
+
+    // un-necessary CComModule::Init call omitted
 }
 
 GameEngine::~GameEngine()
 {
-    // TODO
+    delete g_theMapCache;
+    g_theMapCache = nullptr;
+
+    delete g_theGameResultsQueue;
+
+    g_theSubsystemList->Shutdown_All();
+    delete g_theSubsystemList;
+    g_theSubsystemList = nullptr;
+
+    delete g_theNetwork;
+    g_theNetwork = nullptr;
+
+    delete g_theCommandList;
+    g_theCommandList = nullptr;
+
+    delete g_theNameKeyGenerator;
+    g_theNameKeyGenerator = nullptr;
+
+    delete g_theFileSystem;
+    g_theFileSystem = nullptr;
+
+    delete g_theGameLODManager;
+    Drawable::Kill_Static_Images();
+
+    // un-necessary CComModule::~CComModule call omitted
 
 #ifdef PLATFORM_WINDOWS
     timeEndPeriod(1);
 #endif
 }
 
-void GameEngine::Reset() {}
+void GameEngine::Reset()
+{
+    WindowLayout *layout = g_theWindowManager->Win_Create_Layout("Menus/BlankWindow.wnd");
+    captainslog_dbgassert(layout != nullptr, "We Couldn't Load Menus/BlankWindow.wnd");
+    layout->Hide(false);
+    layout->Bring_Forward();
+    GameWindow *window = layout->Get_Window_List();
+    window->Win_Clear_Status(128);
+    g_theSubsystemList->Reset_All();
+
+    if (g_theGameLogic->Is_In_Multiplayer_Game()) {
```

Comment:
Is_In_Multiplayer_Game needs to be before reset all call

Reply:
fixed.